c/datasetjson.c

```diff
@@ -110,6 +118,16 @@ int streamDataset(Socket *socket, char *filename, int recordLength, jsonPrinter
       }
     }
     fclose(in);
+    rcEtag = icsfDigestFinish(&digest, hash);
+
+    // Convert hash text to hex.
+    char eTag[70] = "";
+    memset(eTag, '\0', strlen(eTag));
```

Comment:
`strlen(eTag)` will return 0. You can just use the init method I mentioned above.